.jenkins/Jenkinsfile

```diff
@@ -11,762 +12,531 @@
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// Se# Licensed to the Apache Software Foundation (ASF) under onee the License for the specific language governing permissions and
+// See the License for the specific language governing permissions and
 // limitations under the License.
 //
 //
-// Jenkins declaration of how to build and test the current codebase.
-//  Jenkins infrastructure related settings should be kept in
-//    https://github.com/apache/cassandra-builds/blob/trunk/jenkins-dsl/cassandra_job_dsl_seed.groovy
+// Jenkins CI declaration.
+//
+// The declarative pipeline is presented first as a high level view.
+//
+// Build and Test Stages are dynamic, the full possible list defined by the `tasks()` function.
+// There is a choice of pipeline profles with sets of tasks that are run, see `pipelineProfiles()`.
+//
+// All tasks use the dockerised CI-agnostic scripts found under `.build/docker/`
+// The `type: test` always `.build/docker/run-tests.sh`
+//
+//
+// This Jenkinsfile is expected to work on any Jenkins infrastructure.
+// The controller should have 4 cpu, 12GB ram (and be configured to use `-XX:+UseG1GC -Xmx8G`)
+// It is required to have agents providing five labels, each that can provide docker and the following capabilities:
+//  - cassandra-amd64-small  : 1 cpu, 1GB ram
+//  - cassandra-small        : 1 cpu, 1GB ram (alias for above but for any arch)
+//  - cassandra-amd64-medium : 3 cpu, 5GB ram
+//  - cassandra-medium       : 3 cpu, 5GB ram (alias for above but for any arch)
+//  - cassandra-amd64-large  : 7 cpu, 14GB ram
+//
+// When running builds parameterised to other architectures the corresponding labels are expected.
+//  For example 'arm64' requires the labels: cassandra-arm64-small, cassandra-arm64-medium, cassandra-arm64-large.
+//
+// Plugins required are:
+//  git, workflow-job, workflow-cps, junit, workflow-aggregator, ws-cleanup, pipeline-build-step, test-stability, copyartifact.
```

Comment:
Do we have to link cassius here?

Reply:
that happens in [CASSANDRA-18145](https://issues.apache.org/jira/browse/CASSANDRA-18145)